hidden-posts.php

```diff
@@ -117,6 +123,35 @@ static function remove_post( $id ) {
         array_splice( $posts, array_search( $id, $posts ), 1 );
 
         update_option( self::META_KEY, array_map( 'intval', $posts ) );
+
+        // Delete post meta.
+        delete_post_meta( $id, self::META_KEY );
+    }
+		
+    /**
+     * Migrate IDs of hidden posts by duplicating them from wp_options to 
+     * wp_postmeta in case the user is using an older version of the plugin.
+     */
+    public function maybe_upgrade_version() {
+      $version = absint( get_option( self::VERSION_KEY, 0 ) );
+
+      if ( $version === self::VERSION ) {
+        return;
+      }
+
+      $posts = self::get_posts();
+
+      if ( ! $posts ) {
+        return;
+      }
```

Comment:
I think you would run `update_option( self::VERSION_KEY, self::VERSION );` here too. If not, `maybe_upgrade_version` is still running till these lines for every `admin_init` action.

Reply:
Good catch. I moved `update_option( self::VERSION_KEY, self::VERSION );` before `$posts = self::get_posts();` to ensure that the script will not run infinitely.